### PR TITLE
Forward port stable branches, Update changelog

### PR DESCRIPTION
**Pull request by micbar**
# Description

Forward port of stable-5.0

### CODE REVIEW

**Comment by mmattel** [comment]
@micbar Can you also add a changlog for 4.0.7, this one is still missing. Maybe a separate PR is necessary?

**Comment by micbar** [comment]
@dragonchaser Can you please review because of the forward ported code changes for the "ShowUserEmailInResults".

File: services/frontend/pkg/config/config.go [hunk 1]
```diff
@@ -144,6 +144,7 @@ type OCS struct {
 	PublicShareMustHavePassword          bool               `yaml:"public_sharing_share_must_have_password" env:"OCIS_SHARING_PUBLIC_SHARE_MUST_HAVE_PASSWORD;FRONTEND_OCS_PUBLIC_SHARE_MUST_HAVE_PASSWORD" desc:"Set this to true if you want to enforce passwords on all public shares." introductionVersion:"5.0"`
 	WriteablePublicShareMustHavePassword bool               `yaml:"public_sharing_writeableshare_must_have_password" env:"OCIS_SHARING_PUBLIC_WRITEABLE_SHARE_MUST_HAVE_PASSWORD;FRONTEND_OCS_PUBLIC_WRITEABLE_SHARE_MUST_HAVE_PASSWORD" desc:"Set this to true if you want to enforce passwords on Uploader, Editor or Contributor shares." introductionVersion:"5.0"`
 	IncludeOCMSharees                    bool               `yaml:"include_ocm_sharees" env:"FRONTEND_OCS_INCLUDE_OCM_SHAREES" desc:"Include OCM sharees when listing sharees." introductionVersion:"5.0"`
+	ShowUserEmailInResults               bool               `yaml:"show_email_in_results" env:"FRONTEND_SHOW_USER_EMAIL_IN_RESULTS;OCIS_SHOW_USER_EMAIL_IN_RESULTS" desc:"Mask user email addresses in responses. (EXPERIMENTAL)" introductionVersion:"5.1"`
```
Comment (dragonchaser) [inline services/frontend/pkg/config/config.go:147]:
```suggestion
	ShowUserEmailInResults               bool               `yaml:"show_email_in_results" env:"OCIS_SHOW_USER_EMAIL_IN_RESULTS" desc:"Mask user email addresses in responses." introductionVersion:"5.1"`
```
- Experimental was for 5.0
- We should opt for global only in that case, there is no use-case where you want to define this per-graph IMHO

File: services/frontend/pkg/config/defaults/defaultconfig.go [hunk 2]
```diff
@@ -116,6 +116,7 @@ func DefaultConfig() *config.Config {
 			ListOCMShares:               true,
 			PublicShareMustHavePassword: true,
 			IncludeOCMSharees:           false,
+			ShowUserEmailInResults:      true,
```
Comment (dragonchaser) [inline services/frontend/pkg/config/defaults/defaultconfig.go:119]:
```suggestion
```

Comment (dragonchaser) [reply services/frontend/pkg/config/defaults/defaultconfig.go:119]: This line is for 5.0 behavior, I think we do not want this in 5.1

File: services/graph/pkg/config/config.go [hunk 3]
```diff
@@ -112,7 +112,7 @@ type API struct {
...
 	AssignDefaultUserRole   bool   `yaml:"graph_assign_default_user_role" env:"GRAPH_ASSIGN_DEFAULT_USER_ROLE" desc:"Whether to assign newly created users the default role 'User'. Set this to 'false' if you want to assign roles manually, or if the role assignment should happen at first login. Set this to 'true' (the default) to assign the role 'User' when creating a new user." introductionVersion:"pre5.0"`
 	IdentitySearchMinLength int    `yaml:"graph_identity_search_min_length" env:"GRAPH_IDENTITY_SEARCH_MIN_LENGTH" desc:"The minimum length the search term needs to have for unprivileged users when searching for users or groups." introductionVersion:"5.0"`
-	ShowUserEmailInResults  bool   `yaml:"show_email_in_results" env:"OCIS_SHOW_USER_EMAIL_IN_RESULTS" desc:"Mask user email addresses in responses." introductionVersion:"5.1"`
+	ShowUserEmailInResults  bool   `yaml:"show_email_in_results" env:"GRAPH_SHOW_USER_EMAIL_IN_RESULTS;OCIS_SHOW_USER_EMAIL_IN_RESULTS" desc:"Mask user email addresses in responses." introductionVersion:"5.1"`
```
Comment (dragonchaser) [inline services/graph/pkg/config/config.go:115]: We should opt for global only in that case, there is no use-case where you want to define this per-graph IMHO
```suggestion
	ShowUserEmailInResults  bool   `yaml:"show_email_in_results" env:"OCIS_SHOW_USER_EMAIL_IN_RESULTS" desc:"Mask user email addresses in responses." introductionVersion:"5.1"`
```

**Review by dragonchaser — APPROVED** [review]

**Review by dragonchaser — CHANGES_REQUESTED** [review]
see comments above (misclicked approve)

**Review by dragonchaser — APPROVED** [review]

**Pull request merged**